internal/app/api/handler/user_handler.go

```diff
@@ -29,29 +29,41 @@ func (h *UserHandler) Login(c fiber.Ctx) error {
 
 	err := c.Bind().Body(req)
 	if err != nil {
-		return sendErrorResp(c, fiber.StatusBadRequest, constants.ErrMsgInvalidUsernameAndPassword)
+		return sendErrorResponse(c, fiber.StatusBadRequest, constants.ErrMsgInvalidUsernameAndPassword)
 	}
 
 	if req.Username == "" || req.Password == "" {
-		return sendErrorResp(c, fiber.StatusBadRequest, constants.ErrMsgUsernameOrPasswordRequired)
+		return sendErrorResponse(c, fiber.StatusBadRequest, constants.ErrMsgUsernameOrPasswordRequired)
 	}
 
 	userLoginData, err := h.userService.Login(ctx, req.Username, req.Password)
 	if err != nil {
 		if err == sql.ErrNoRows {
-			return sendErrorResp(c, fiber.StatusUnauthorized, constants.ErrMsgUsernameNotFound)
+			return sendErrorResponse(c, fiber.StatusUnauthorized, constants.ErrMsgUsernameNotFound)
 		}
 
-		return sendErrorResp(c, fiber.StatusUnauthorized, constants.ErrMsgInvalidUsernameOrPassword)
+		return sendErrorResponse(c, fiber.StatusUnauthorized, constants.ErrMsgInvalidUsernameOrPassword)
 	}
 
-	return sendSuccessResp(c, fiber.StatusOK, "Success", userLoginData)
+	return sendSuccessResponse(c, fiber.StatusOK, "Success", userLoginData)
 }
 
 func (h *UserHandler) Register(c fiber.Ctx) error {
-	span, _ := opentracing.StartSpanFromContext(c.Context(), "handler.UserHandler.Register")
+	span, ctx := opentracing.StartSpanFromContext(c.Context(), "handler.UserHandler.Register")
 	defer span.Finish()
 
+	req := &registerRequest{}
+	err := c.Bind().Body(req)
+	if err != nil {
+		return sendErrorResponse(c, fiber.StatusBadRequest, "Invalid request body. Please provide name , username, password, confirmpassword,email.")
+	}
 	// TODO
+
+	bool, err := h.userService.Register(ctx, req.Username, req.Email, req.Password, req.ConfirmPassword, req.Name)
+	if bool {
+		return sendSuccessResponse(c, fiber.StatusOK, "Account Added Succesfully.", "DONE!!")
```

Comment:
Do not put something uneceserry on response data, just set data as nil instead of "DONE!!"